### DISCUSSION TITLE
Suggestions for Bottles Integration

### DISCUSSION THREAD

**Discussion by ManedCalico** [PhilipK/BoilR#243] [Ideas]
Hey there!

First off, I made a GitHub account just to write this! Sorry if I'm not doing this properly... please be gentle! I'm really thankful for Boilr, it's really helped make it so easy to add artwork and make my Deck pretty! I do have some suggestions to make its recent update to include Bottles support smoother. Hopefully this is the right place to put this!

So I've installed all third-party launchers through Bottles, and I've noticed some quirks. From what I can tell, Boilr pulls the games it sees in Bottles from Bottles' Programs list rather than scanning the physical files within the Bottles path. However, Bottles has some issues with how it pulls in games, as well as sometimes being unable to remove or rename them properly. This is causing issues with how Boilr handles games in Bottles.

Here's what I'm noticing:

1. EGS games in Bottles do not show up in Boilr when Bottles automatically adds them to the Programs list. However, trying to rename an EGS game in the Program list creates a renamed duplicate entry instead. Once this duplicate is created, both the original and the duplicate show up in Boilr. Deleting the duplicate makes both disappear. If Bottles auto-adds the game with the correct name, just clicking rename and save doesn't work since it doesn't create the duplicate. So something like "Control" needs to be made incorrect like "Control!" to work.

2. Continuing from the above, the original non-renamed program cannot be removed from the Bottles Program list. However, it can be hidden. Unfortunately, Boilr still shows hidden entries. This is also true of launchers that install sub-processes, like Origin's webhelper service, and Battle.net that shows up in Bottles as both "Battle.net" and "Battle". Bottles won't allow them to be removed, and Boilr still shows them if they're hidden. It know I can uncheck them, but with the issue above it means every EGS game has to have two items in Boilr, one checked and one unchecked. It would be great if Boilr ignored hidden entries!

3. If Boilr doesn't see a game and you instead use Bottles' "Add to Steam" feature, it causes some really odd things to happen! If you try to edit the images in Boilr, it modifies all of the shortcuts created by Bottles together. They're listed separately with the correct names, but the SteamGridDB ID for all of them is the same. If you change the ID and pick some art for one, it's propagated across all of them in Boilr. Also, picking art doesn't actually apply the art to the shortcut. Restarting Steam or returning to Gaming mode still shows the grey defaults.

4. A general suggestion-- I have each launcher installed in its own bottle, and it would be nice to have Boilr show the games per bottle in Bottles instead of one big Bottles section. Like "Bottles: itch", "Bottles: Origin", etc based on the bottle name.

5. Another suggestion continuing from the above-- it would be nice to have Boilr save the list of found/managed games between sessions, and have a "refresh" button for each section. So it wouldn't need to find all of the same games again on startup if it remembers what its managed previously. If I install new games, I could just open Boilr and refresh the specific section to have it find new entries. This would help with large libraries.

That's it! Sorry if this is a lot for one post. Just trying to be helpful! I'm also new to Linux, so maybe some of these things are just me doing something wrong... if so, I apologize.

Thanks!

**Comment by PhilipK** [comment]
First of, yes, you are doing things right :-) Don't worry.
Thanks for your feedback. I am at work, but will look at this later.

**Comment by PhilipK** [comment]
Alright, let me see if I can answer all of these.

I have not played around too much with bottles myself, so I have not seen some of the behavior that you mention. But I can try to explain what BoilR does and maybe it will then make sense why it is acting weird.

BoilR calls the Bottles CLI, and the CLI lists all the bottles and the programs in there.
BoilR does not look into the bottles , it totally trusts what the Bottles CLI tells it.
Also, Bottles is just one of the "Platforms" that BoilR supports, that is why the games are shown in only one list, even if they are from different launchers inside Bottles.

Maybe I will have to rethink how the bottles integration works, because it seems like the domain/mental model that has fit the other launchers does not really fit with Bottles.

**Reply by ManedCalico** [reply]
Hey Philip!

Thank you so much for replying! The explanation is really helpful. I've been tinkering with and reading up on Bottles more, and knowing how BoilR gets its information is making me realize a few things.

I think I've figured out why Bottles was behaving so oddly with EGS, and this might also explain why games aren't showing up consistently in Boilr. Normally Bottles searches for the Start Menu shortcuts created inside a bottle to add to the Programs list. This seems to have been the cause of why I was seeing two Battle.net and Origin entires, those installs created multiple shortcuts. Deleting the extra .lnk files in a bottle removed them. No need for hiding them anymore!

However, I couldn't find any shortcuts created by ESG and I couldn't even force it to manually create them on the desktop, but I did find a setting in preferences for automatically adding EGS games, which was enabled by default. Toggling the setting makes the misnamed EGS entires disappear and reappear immediately without rescanning a bottle.

What I think is happening is that Bottles is maybe reading the games list directly from the EGS launcher somehow instead of searching for shortcuts. Games added this way aren't seen by Boilr and they can't be deleted or renamed in Bottles. Trying to rename them in Bottles creates a new entry instead, probably in a similar way to adding a shortcut manually.

I ran into other instances of items that were automatically added by Bottles that wouldn't show up in BoilR either, but anything added manually always shows up. So it might be a Bottles CLI issue that makes it not include an item if it's added automatically this way?

As far as the oddness with Bottles' "Add to Steam" function... the way it's creating shortcuts is different than Boilr. Boilr's non-game shortcuts work 100% of the time, but I found that Bottles shortcuts are a bit more broken than I originally thought. I wouldn't spend time looking into it, since Boilr is much more reliable!

I didn't mean to cause too much of a headache! That definitely makes sense that Boilr sees Bottles as one platform of many, similar to how Heroic is both Epic and GOG. Having the games grouped by bottle would be handy if there were a way to refresh the games list manually per platform or bottle, but since it refreshes all at once there isn't much of a reason to get granular like that. Also, being new to all this, I'm not even sure if it makes sense to have a bottle for each platform or if I should've just had one bottle for everything. So it's probably a me-issue more than anything else!

Thank you again for taking the time to think through this with me!
